### PR TITLE
feature: combine root and nonroot secret install; delay chowning

### LINKED ISSUE #117: Order of operations
As suggested in Matrix I am creating this issue about the order in which agenix operates.

When switching generations I consistantly see the following order of operations in the log output:
```
[agenix] symlinking new secrets to /run/agenix (generation 6)...
[agenix] removing old secrets (generation 5)...
[agenix] decrypting root secrets...
decrypting '/nix/store/xxx.age' to '/run/agenix.d/6/xxx'...
decrypting '/nix/store/yyy.age' to '/run/agenix.d/6/yyy'...
[agenix] decrypting non-root secrets...
decrypting '/nix/store/zzz.age' to '/run/agenix.d/6/zzz'...
```

This got me wondering as I would expect the order of operations to be:
1) decrypt secrets into new generation
2) symlink new secrets
3) remove old generation/secrets

Could we get some clarification about the actual order?

### PR DESCRIPTION
**Pull request by ryantm**
This simplifies agenix by combining the root and nonRoot secret installation into one place and delays setting the owner and group of the secrets until after the users and groups activation scripts are completed. This also fixes #117 by incorporating the changes from #118 to not switch over the secret directory symlink until after the secrets are decrypted. 

This is a **breaking change** in the sense that someone might have depended on the user or group of a root secret being set before the "users" and "groups" activation scripts run, but that seems unlikely to me.

This is also a **breaking change** because I renamed a bunch of the activation scripts. I believe these will be module compile-time errors.

cc @jsimonetti

### CODE REVIEW

**Comment by jsimonetti** [comment]
Perhaps it would be wise (considering the possible breaking) to tag a new release before and after merging?

**Review by cole-h — APPROVED** [review]
Sorry for letting this hang so long -- this looks fine to me, but I haven't tested it.

**Comment by cole-h** [comment]
We should be sure to cut a release before and after this merges, though.

**Comment by jsimonetti** [comment]
Just curious, is there anything blocking this PR?

**Comment by ryantm** [comment]
Both those releases have been cut and release notes added. Thanks for the reviews everyone!

**Pull request merged**
